src/lib.rs

```diff
@@ -70,6 +77,128 @@ fn hash_with_seed<T: Hash>(iter: u64, v: &T) -> u64 {
     state.finish()
 }
 
+pub trait FastIteration {
```

Comment:
Interesting, how much of speedup does this give?  I'm a concerned that we're now requiring the user of this library to implement a new trait to use it, which makes it a lot more difficult to start using.  Can we have a special version of the Mphf constructor that uses this?  Or is there a way to have a default implementation of this trait for regular iterators, that specialized clients can overrride?

Reply:
See if you can put a 'default' impl of FastIteration into this crate, and still override with your specialized version for the DBG. it will look something like this:
```
impl<T> FastIteration for T: Iterator {
  fn skip_next(&mut self) {
      self..next();
   }
}
```

I'm not 100% sure this will work... maybe related to the concept of 'specialization' in rust...

Reply:
Got answer from @shepmaster [here](https://stackoverflow.com/questions/51235721/how-can-a-function-conditionally-fall-back-to-a-trait-if-another-trait-is-implem/51237640#51237640).

Reply:
You were right about the `specialization` feature !